tests/fsharpqa/Source/Conformance/DeclarationElements/MemberDefinitions/OptionalDefaultParameterValueArguments/ConsumeFromCS.fs

```diff
@@ -0,0 +1,42 @@
+﻿// #Conformance #DeclarationElements #MemberDefinitions #OptionalDefaultParameterValueArguments 
+
+namespace ConsumeFromCS
+
+open System.Runtime.InteropServices
+open System
+
+type Class() =
+    //check that all known supported primitive types work.
+    static member Method1  ([<Optional;DefaultParameterValue(42y)>]i:sbyte) = i
+    static member Method2  ([<Optional;DefaultParameterValue(42uy)>]i:byte) = i
+    static member Method3  ([<Optional;DefaultParameterValue(42s)>]i:int16) = i
+    static member Method4  ([<Optional;DefaultParameterValue(42us)>]i:uint16) = i
+    static member Method5  ([<Optional;DefaultParameterValue(42)>]i:int32) = i
+    static member Method6  ([<Optional;DefaultParameterValue(42u)>]i:uint32) = i
+    static member Method7  ([<Optional;DefaultParameterValue(42L)>]i:int64) = i
+    static member Method8  ([<Optional;DefaultParameterValue(42UL)>]i:uint64) = i
+    static member Method9  ([<Optional;DefaultParameterValue(42.42f)>]i:single) = i
+    static member Method10 ([<Optional;DefaultParameterValue(42.42)>]i:double) = i
+    static member Method11 ([<Optional;DefaultParameterValue(true)>]i:bool) = i
+    static member Method12 ([<Optional;DefaultParameterValue('q')>]i:char) = i
+    static member Method13 ([<Optional;DefaultParameterValue("ono")>]i:string) = i
+    //Check a reference type - only null possible.
+    static member Method14 ([<Optional;DefaultParameterValue(null)>]i:obj) = i
+    //Check a value type - only default ctor possible.
+    static member Method15 ([<Optional;DefaultParameterValue(new DateTime())>]i:DateTime) = i
+
+    //Check nullables. 
+    static member MethodNullable1 ([<Optional;DefaultParameterValue(null)>]i:Nullable<int>) = i
```

Comment:
Yes, or whatever the default value of the type is. To support that we could add some type-directed analysis (a simplified version of `analyze` which you mentioned earlier - wouldn't need to support marshal and ref parameters) to the added code in infos.fs.